httpcore/_async/http11.py

```diff
@@ -72,13 +74,13 @@ async def arequest(
         ) = await self._receive_response(timeout)
         response_stream = AsyncIteratorByteStream(
             aiterator=self._receive_response_data(timeout),
-            aclose_func=self._response_closed,
         )
         ext = {
             "http_version": http_version.decode("ascii", errors="ignore"),
             "reason": reason_phrase.decode("ascii", errors="ignore"),
         }
-        return (status_code, headers, response_stream, ext)
+        yield (status_code, headers, response_stream, ext)
+        await self._response_closed()
```

Comment:
@tomchristie Not sure this comes from you, do we need a `try/finally` here?

```suggestion
        try:
            yield (status_code, headers, response_stream, ext)
        finally:
            await self._response_closed()
```

Reply:
I'd thought it wasn't required, but since your comment I've convinced myself that I was wrong, using this...

```python
from contextlib import contextmanager


@contextmanager
def request(method, url):
    print('request() started')
    try:
        yield 200, {}, [b'Hello, world',]
    finally:
        print('request() cleaning up')


def main():
    print('main')
    with request("GET", "https://example.org/") as response:
        raise ValueError('Oops')


main()
```

Versus this...

```python
from contextlib import contextmanager


@contextmanager
def request(method, url):
    print('request() started')
    yield 200, {}, [b'Hello, world',]
    print('request() cleaning up')  # Never runs


def main():
    print('main')
    with request("GET", "https://example.org/") as response:
        raise ValueError('Oops')


main()
```